### PR TITLE
DEV: allow partial uploads in conda_build_matrix

### PR DESCRIPTION
**Pull request by llllllllll**
I cannot get the builds to run in parallel without locking up. We can invest more time if this becomes an annoyance

### CODE REVIEW

**Comment by richafrank** [comment]
> I cannot get the builds to run in parallel without locking up. 

Since this is dev-only, maybe it would be easy to get it working on a single python version.

File: etc/conda_build_matrix.py [hunk 1]
```diff
@@ -74,11 +80,21 @@ def main(upload, upload_only, user):
...
+            out, err = p.communicate()
+            if p.returncode:
+                # only change the status to failure if we are not allowing
+                # partial uploads
+                status |= not allow_partial_uploads
+                print(
+                    'failed to get the output name for python=%s numpy=%s\n'
+                    '%s' % (py_version, npy_version, err.decode('utf-8')),
+                )
+            files.append(out.decode('utf-8').strip())
```
Comment (richafrank) [inline etc/conda_build_matrix.py]: Should this be inside the `p.returncode` conditional?

File: etc/conda_build_matrix.py [hunk 2]
```diff
@@ -74,11 +80,21 @@ def main(upload, upload_only, user):
                     err.decode('utf-8'),
                 ))
         elif upload:
-            files.append(subprocess.Popen(
+            p = subprocess.Popen(
                 **mkargs(py_version, npy_version, output=True)
```
Comment (richafrank) [inline etc/conda_build_matrix.py]: How does this differ from the functionality of `proc`?  Are we `conda build`ing twice?

File: etc/conda_build_matrix.py (hunk 1 above)
Comment (llllllllll) [inline etc/conda_build_matrix.py]: yup

File: etc/conda_build_matrix.py (hunk 2 above)
Comment (llllllllll) [inline etc/conda_build_matrix.py]: nope, this just prints the filename that the last command built

Comment (richafrank) [inline etc/conda_build_matrix.py]: Ahhh `--output`

File: etc/conda_build_matrix.py [hunk 3]
```diff
@@ -43,11 +43,17 @@ def mkargs(py_version, npy_version, output=False):
     help='Upload the last built packages without rebuilding.',
 )
 @click.option(
+    '--allow-partial-uploads',
+    is_flag=True,
+    default=False,
+    help='Allow uploads',
```
Comment (richafrank) [inline etc/conda_build_matrix.py]: Could you include that this only works with `--update-only`?

File: etc/conda_build_matrix.py [hunk 4]
```diff
@@ -74,11 +80,21 @@ def main(upload, upload_only, user):
...
+            out, err = p.communicate()
+            if p.returncode:
+                # only change the status to failure if we are not allowing
+                # partial uploads
+                status |= not allow_partial_uploads
+                print(
+                    'failed to get the output name for python=%s numpy=%s\n'
+                    '%s' % (py_version, npy_version, err.decode('utf-8')),
+                )
+                files.append(out.decode('utf-8').strip())
```
Comment (richafrank) [inline etc/conda_build_matrix.py]: Actually, this should be in an `else`, right?

Comment (llllllllll) [inline etc/conda_build_matrix.py]: I am struggling ;_;, thanks

File: etc/conda_build_matrix.py (hunk 3 above)
Comment (llllllllll) [inline etc/conda_build_matrix.py]: this should work without upload only. I will fix this because it should be more descriptive

Comment (richafrank) [inline etc/conda_build_matrix.py]: Ah, I see - that's the second loop.

**Pull request merged**